### PR TITLE
Stop requiring a bleeding-edge Cython.

### PR DESCRIPTION
**Pull request by alexmv**
Fixes: #204.  See #205 for an alternate fix, which I don't prefer.

### CODE REVIEW

**Comment by alexmv** [comment]
For any future visitors to this PR, most of the discussion was on #205.

**Comment by Erotemic** [comment]
Yikes, it looks like you have some auto-formatting enabled. That makes the diff a lot more complex than it needs to be. Is there any way to revert that?

I absolutely open to formatting PRs, but it would be good if it was separated into a different PR. I also I have strong opinions on quotes. Particularly I believe:

    * Single quotes (') are used for code
    * Double quotes (") are used for documentation
    * Don't touch any string that has an internal quote.
    
I also do have an [auto-formatter that accomplishes this](https://github.com/Erotemic/xdev/blob/main/xdev/format_quotes.py).

I'm fine with the changes to the docstring spacing, import sorting, and getting rid of parens around the yeild, but the extra new lines in the argument parser and other function calls and switching to double quotes is something I'd rather not have.

**Comment by alexmv** [comment]
> Yikes, it looks like you have some auto-formatting enabled. That makes the diff a lot more complex than it needs to be. Is there any way to revert that?

Yikes indeed!  Sorry, that was absolutely unintentional -- I have a hook to apply [black](https://github.com/psf/black), which fired.  I've repushed with those formatting changes dropped.

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=41.0.1", "Cython>=3.0.0a11"]
+requires = ["setuptools>=41.0.1", "Cython>=0.29.24,<=3.0.0a11"]
```
Comment (andersk) [inline pyproject.toml:2]: For Python 3.12 compatibility, should this use the same environment markers you wrote in `requirements/build.txt`? (They are supported here: see [PEP 518](https://peps.python.org/pep-0518/#build-system-table), [PEP 508](https://peps.python.org/pep-0508/#environment-markers).)

Comment (Erotemic) [reply pyproject.toml:2]: Yes, that would be great.

Comment (alexmv) [reply pyproject.toml]: TIL, as well.  Repushed with those.

**Comment by Erotemic** [comment]
Thanks @alexmv @andersk @Theelx ! Merging and releasing.

**Pull request merged**